### PR TITLE
Fix minimize failing for generic interface TypeSpecs with no member refs

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Fix FillTypeSpecsFromTypes() for not scanning interface implementations.
- RemoveEmptyItems() was removing TypeSpecs with no member refs without checking whether the TypeSpec is used as an interface type.
- Dumper generator now properly deals with typespecs without member references.
- Add code to test app to cover these situations.

## Motivation and Context
- Edge case surface when working on adding `Dictionary<T,V>` to System.Collections.
- Related with nanoframework/Home#782.
- [tested against nanoclr buildId 61101].

## How Has This Been Tested?
- Added new code to test this in TestNFApp project.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [x] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by josesimoes** [comment]
@coderabbitai review

File: MetadataProcessor.Tests/TestNFApp/GenericInterfaceTypeSpecTests.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            if (_count == _items.Length)
+            {
+                var bigger = new Pair<TKey, TValue>[_items.Length * 2];
+                for (int i = 0; i < _count; i++) bigger[i] = _items[i];
+                _items = bigger;
+            }
+
+            _items[_count++] = new Pair<TKey, TValue>(key, value);
+        }
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply MetadataProcessor.Tests/TestNFApp/GenericInterfaceTypeSpecTests.cs:61]: Done

File: MetadataProcessor.Tests/TestNFApp/GenericInterfaceTypeSpecTests.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            Key = key;
+            Value = value;
+        }
+    }
+
+    // An interface that inherits IEnumerable<Pair<TKey,TValue>>.
+    // The C# compiler adds IEnumerable<Pair<TKey,TValue>> to InterfaceImpls —
+    // a GenericInstanceType TypeSpec with no member references in this assembly.
+    public interface IPairCollection<TKey, TValue> : IEnumerable
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply MetadataProcessor.Tests/TestNFApp/GenericInterfaceTypeSpecTests.cs]: Fixed

File: MetadataProcessor.Shared/Tables/nanoTypeSpecificationsTable.cs [hunk 3]
```diff
@@ -609,6 +621,20 @@ internal void RemoveEmptyItems()
...
+                        string genericInstanceFullName = genericInstanceType.FullName;
+                        foreach (TypeDefinition td in _context.TypeDefinitionTable.Items)
+                        {
+                            if (td.HasInterfaces && td.Interfaces.Any(i => i.InterfaceType.FullName == genericInstanceFullName))
+                            {
+                                hasMemberReferences = true;
+                                break;
+                            }
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply MetadataProcessor.Shared/Tables/nanoTypeSpecificationsTable.cs:637]: Will be addressed on a separate PR.

File: MetadataProcessor.Shared/nanoDumperGenerator.cs [hunk 4]
```diff
@@ -799,7 +799,14 @@ private void DumpTypeSpecifications(
...
 
-                Debug.Assert(typeSpec.MemberReferences.Count > 0, $"Couldn't find any MethodRef for TypeSpec[{typeReference}] {typeReference.FullName}");
+                if (typeSpec.MemberReferences.Count == 0)
+                {
+                    // A TypeSpec used only as an interface implementation has no member references — that is valid.
+                    bool usedAsInterface = _tablesContext.TypeDefinitionTable.Items
+                        .Any(td => td.Interfaces.Any(i => i.InterfaceType.FullName == typeReference.FullName));
+
+                    Debug.Assert(usedAsInterface, $"Couldn't find any MethodRef for TypeSpec[{typeReference}] {typeReference.FullName}");
+                }
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply MetadataProcessor.Shared/nanoDumperGenerator.cs:809]: Will be addressed on a separate PR.
